### PR TITLE
[confmap.provider/s3provider] Support additional config sources for `s3provider` other than environment

### PR DESCRIPTION
**Pull request by neelayu**
**Description:** <Describe what has changed.>
Support additional config sources for `s3provider`. Also fixed few code comments, readme and added couple of unit tests

**Link to tracking Issue:**  #14967

**Testing:** NA

**Documentation:** Added readme

Part of PR split #14617

### CODE REVIEW

File: confmap/provider/s3provider/README.md
Comment (neelayu) [reply confmap/provider/s3provider/README.md]: Thanks for the comment. I have made the suggestions requested and also added support for those configurations.

**Comment by neelayu** [comment]
not stale?

File: confmap/provider/s3provider/provider.go
Comment (Aneurysm9) [inline confmap/provider/s3provider/provider.go:57]: At this time we do not have any config providers that accept parameters on initialization.  We also don't have a formalized method for including new config providers in the collector builder and I worry that taking arbitrary parameters in the initializer could complicate doing so.  

@neelayu can you please provide some more context on why this is required and the existing environment-based configuration of the underlying AWS SDK is insufficient?

@open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers I'd like your feedback on this.

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]

File: confmap/provider/s3provider/provider.go
Comment (neelayu) [reply confmap/provider/s3provider/provider.go:57]: Your point is valid. Although the current state of the collector builder(ocb) has no support for custom providers as far as I know.(of course that can change). This would generally be used by distributions not using the builder.

@bryan-aguilar suggested that there are other ways of configuring the creds source. I thought of extending that support in the provider.

**Comment by neelayu** [comment]
not stale

**Review by mx-psi — DISMISSED** [review]
Marking this as `on hold` until we figure out how to support configuring confmap providers. I opened open-telemetry/opentelemetry-collector/issues/6956 to discuss this.

**Comment by mx-psi** [comment]
Removing `on hold` and block since https://github.com/open-telemetry/opentelemetry-collector/issues/6956 has been resolved

**Pull request closed without merging**
